# Detecção de Idioma


Instalação da biblioteca langdetect

In [ ]:
!pip install langdetect

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 981.5/981.5 kB 8.7 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Created wheel for langdetect: filename=langdetect-1.0.9-py3-none-any.whl size=993227 sha256=7e493a273eacf4a103d09fccf4d9b7ce9c3a243f1b8e3d32c553ae2dda9a0891
  Stored in directory: /root/.cache/pip/wheels/95/03/7d/59ea870c70ce4e5a370638b5462a7711ab78fba2f655d05106
Successfully built langdetect


Importação da biblioteca langdetect

In [ ]:
from langdetect import detect

Frase

In [ ]:
frase = "おはよう"

Detectar o idioma da frase

In [ ]:
idioma = detect(frase)

Printar o Idioma

In [ ]:
print("O idioma da frase é:", idioma)

O idioma da frase é: ja


# Analisar o sentimento de uma frase


## Sem biblioteca

Frase

In [ ]:
frase = "Estou muito feliz com a notícia."

Convertendo a frase para minúsculas para facilitar a comparação

In [ ]:
frase = frase.lower()

Definindo palavras-chave positivas e negativas

In [ ]:
positive_words = ["feliz", "alegre", "ótimo", "maravilhoso"]
negative_words = ["triste", "deprimido", "terrível", "horrível"]

Inicializando a variável de sentimento

In [ ]:
sentiment = 0

Verificando se alguma palavra-chave positiva está presente na frase

In [ ]:
for word in positive_words:
    if word in frase:
        sentiment = 1  # Sentimento positivo
        break

Verificando se alguma palavra-chave negativa está presente na frase

In [ ]:
for word in negative_words:
    if word in frase:
        sentiment = -1  # Sentimento negativo
        break

Determinando o sentimento final

In [ ]:
if sentiment == 1:
    sentiment_label = "positivo"
elif sentiment == -1:
    sentiment_label = "negativo"
else:
    sentiment_label = "neutro"

Exibindo o resultado

In [ ]:
print("Frase:", frase)
print("Polaridade do sentimento:", sentiment)
print("Sentimento:", sentiment_label)

Frase: estou muito feliz com a notícia.
Polaridade do sentimento: 1
Sentimento: positivo


## Com a biblioteca Transformes

In [ ]:
!pip install transformers

from transformers import BertTokenizer, BertForSequenceClassification
from transformers import pipeline

# Carregar o modelo BERT pré-treinado para análise de sentimento
model_name = "nlptown/bert-base-multilingual-uncased-sentiment"
model = BertForSequenceClassification.from_pretrained(model_name)
tokenizer = BertTokenizer.from_pretrained(model_name)

# Criar um pipeline para análise de sentimento
nlp_pipeline = pipeline("sentiment-analysis", model=model, tokenizer=tokenizer)



/usr/local/lib/python3.10/dist-packages/huggingface_hub/utils/_token.py:89: UserWarning: 
The secret `HF_TOKEN` does not exist in your Colab secrets.
To authenticate with the Hugging Face Hub, create a token in your settings tab (https://huggingface.co/settings/tokens), set it as secret in your Google Colab and restart your session.
You will be able to reuse this secret in all of your notebooks.
Please note that authentication is recommended but still optional to access public models or datasets.
  warnings.warn(


config.json:   0%|          | 0.00/953 [00:00<?, ?B/s]

pytorch_model.bin:   0%|          | 0.00/669M [00:00<?, ?B/s]

tokenizer_config.json:   0%|          | 0.00/39.0 [00:00<?, ?B/s]

vocab.txt:   0%|          | 0.00/872k [00:00<?, ?B/s]

special_tokens_map.json:   0%|          | 0.00/112 [00:00<?, ?B/s]

In [ ]:
# Frase de exemplo
frase = "Estou muito feliz hoje!"

# Analisar o sentimento da frase usando o modelo BERT
resultado = nlp_pipeline(frase)

# Exibir o resultado
print("Frase:", frase)
print("Sentimento:", resultado)

Frase: Estou muito feliz hoje!
Sentimento: [{'label': '5 stars', 'score': 0.7578408122062683}]


In [ ]:
# Frase de exemplo
frase2 = "Estou normal hoje"

# Analisar o sentimento da frase usando o modelo BERT
resultado2 = nlp_pipeline(frase2)

# Exibir o resultado
print("Frase:", frase2)
print("Sentimento:", resultado2)


Frase: Estou normal hoje
Sentimento: [{'label': '3 stars', 'score': 0.364665687084198}]


A frase fornecida é "Estou muito feliz hoje!".

O resultado da análise de sentimento é uma lista de dicionários, onde cada dicionário contém duas chaves: "label" e "score".

O "label" indica a classificação do sentimento. Neste caso, a classificação é "5 stars", o que indica um sentimento muito positivo.

O "score" é um valor entre 0 e 1 que indica a confiança do modelo na classificação atribuída. Neste caso, o modelo atribuiu um score de aproximadamente 0.758, o que sugere uma confiança moderadamente alta na classificação de "5 stars" para o sentimento da frase fornecida.